### ISSUE TITLE
enlightenment: module uses import-from-derivation

### ISSUE THREAD

**Issue by emilazy** [NixOS/nixpkgs#87678]
The Enlightenment derivation [patches the build](https://github.com/NixOS/nixpkgs/blob/11f49fb94d39ea4548d019542b0d7739d7019789/pkgs/desktops/enlightenment/enlightenment.suid-exes.patch) to symlink paths in the store into `/run/wrappers/bin` and generate a NixOS configuration file with `security.wrappers` configuration, and then [uses import-from-derivation](https://github.com/NixOS/nixpkgs/blob/11f49fb94d39ea4548d019542b0d7739d7019789/nixos/modules/services/x11/desktop-managers/enlightenment.nix#L70) to pull it into the system configuration. This is not a great idea for a bunch of reasons, but the most relevant one right now is that the newly-added test from #86519 doesn't work on Hydra (see https://github.com/NixOS/nixpkgs/issues/87592#issuecomment-627391267).

There are only five wrappers in the generated file:

```
"enlightenment_backlight.orig".source = "/nix/store/s84q2hf44n3g5qmw4n18cy7gfpp4fim2-enlightenment-0.23.1/lib/enlightenment/utils/enlightenment_backlight.orig";
"enlightenment_sys.orig".source = "/nix/store/s84q2hf44n3g5qmw4n18cy7gfpp4fim2-enlightenment-0.23.1/lib/enlightenment/utils/enlightenment_sys.orig";
"enlightenment_ckpasswd.orig".source = "/nix/store/s84q2hf44n3g5qmw4n18cy7gfpp4fim2-enlightenment-0.23.1/lib/enlightenment/utils/enlightenment_ckpasswd.orig";
"freqset.orig".source = "/nix/store/s84q2hf44n3g5qmw4n18cy7gfpp4fim2-enlightenment-0.23.1/lib/enlightenment/modules/cpufreq/linux-gnu-x86_64-0.23.1/freqset.orig";
"cpuclock_sysfs.orig".source = "/nix/store/s84q2hf44n3g5qmw4n18cy7gfpp4fim2-enlightenment-0.23.1/lib/enlightenment/modules/sysinfo/linux-gnu-x86_64-0.23.1/cpuclock_sysfs.orig";
```

so I think that the best approach is to hardcode these in the NixOS module in the standard way and remove the setuid install script entirely. Unfortunately, the paths to these binaries are [hardcoded in other Enlightenment binaries](https://github.com/NixOS/nixpkgs/blob/11f49fb94d39ea4548d019542b0d7739d7019789/pkgs/desktops/enlightenment/enlightenment.nix#L50-L63), so we can't simply install the wrappers in PATH and have everything work out; the Enlightenment build system or source code will need to be patched to look for these helper binaries in `/run/wrappers/bin` instead, which is where my expertise runs out.

(All that said, I really wonder if we should be installing 5 setuid binaries for a desktop environment like this; does Enlightenment have no support for polkit/logind/...?)

cc @romildo

**Comment by romildo** [comment]
A new release of Enlightenment will happen very soon. The [Enlightenment DR 0.24.0-beta1 Release announcement](https://www.enlightenment.org/news/e24.0-beta1) highlights:

> * Reduced number of setuid tools (merged many into single system tool)

There are some changes regarding setuid programas discussed in the thread [[E-devel] Updates from E Development land](https://sourceforge.net/p/enlightenment/mailman/message/36918076/) from their development mailing list:

>  this requires the revamp i did of the setuid root tools and merging them all into
one root service daemon e spawns to do these things

> Setuid helper rewrite
> ---------------------
> 
> I merged all the functionality of the various setuid root tools e uses for
things like /sys/ fs backlight control etc. into a single daemon that hangs
about and e talks to via stdin/out. This reduces the number of these and has a
single access point now. This has replaced everything except the eeze mount
helper stuff in EFM, the new tool can replace the mounting/unmounting needs
and the code is there, but the way eeze works means I can't slot it in easily.
I have to rewrite the way EFM works to do this, but this is a nice "invisible
cleanup" that should provide a single gateway for permissions on who can run
this tool etc. and was necessary for the DDC stuff to not be glacially slow.

[Various security issues in setuid-root binary `enlightenment_system`](https://sourceforge.net/p/enlightenment/mailman/message/36994273/) has been reported but they have already been fixed.

My suggestion is to fix this issue starting with the [Enlightenment DR 0.24.0-beta1 Release](https://www.enlightenment.org/news/e24.0-beta1).

**Comment by emilazy** [comment]
To be honest, the reaction to security reports in that thread makes me even more wary to be installing this binary as setuid root silently in a module...

But hopefully it will simplify the derivation and allow the test to avoid IFD.

**Comment by ajs124** [comment]
> A new release of Enlightenment will happen very soon.

The release has happened now and was picked up by r-ryantm in #88487

**Comment by wirew0rm** [comment]
This looks like it was resolved by #88603 and should probably be closed.

**Comment by cole-h** [comment]
Indeed, looks like it was resolved by the aforementioned PR.
